### DISCUSSION TITLE
Toggle between complex multi-frame layouts with a single frame max layout

### DISCUSSION THREAD

**Discussion by wdkrnls** [herbstluftwm/herbstluftwm#1549] [Ideas]
My screen is small yet I often have client's which display long paths or URLs in a file manager. These paths are clickable, producing new clients. Unless the frame I am in is big, I cannot read the "links". Right now it's easy to toggle a client fullscreen, so I use fullscreen a lot to study these "links". However, fullscreen isn't really what I want since it doesn't really play nicely with other clients. Once focus is changed the fullscreen window just gets in the way. I would much prefer to toggle between the current frame configuration on the tag and a single frame with all clients with max layout. Then I can use already available tools to quickly jump between windows. Afterwards, I would like to go back to the spatially more complex configuration. 

Most of the building blocks are there to write something like what I'm describing with a script. You can dump and load the current configuration. You can query all the clients on a tag. You can bring them all to the focused frame. I'm ignorant of a good way to iterate through and close all empty frames afterwards. I'm also not sure how to implement the toggle. Presumably,  I need to write the previous configuration to a tag-specific state file and write a script which loads that file if the tag is in the single frame max layout I describe. Afterwards, any newly created windows would be sent to the previously focused frame.

**Comment by t-wissmann** [comment]
The `maximize.sh` script ([in the repo](https://github.com/herbstluftwm/herbstluftwm/blob/master/scripts/maximize.sh)) implements precisely what you are trying to accomplish :-)

**Reply by wdkrnls** [reply]
Thanks! It works perfectly as you said :-)
